### PR TITLE
165 new bridge UI

### PR DESCRIPTION
**Pull request by L03TJ3**
# Description
This PR is for updating the UI for the micro-bridge, and fixing existing bugs:

- [x] - use explorer to fetch balances
- [x] - fix input/ouput validation
- [x] - fix input/output calculations
- [x] - update `MicroBridge` to align with new requested UI
- [x] - Add recent transactions list 

Extra:
- [x] some pending/not released goodid theming and components are included 

Todo:
- [x] Slighly altered design and copies have to be confirmed

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [ ] PR title matches follow: (Feature|Bug|Chore) Task Name
- [ ] My code follows the style guidelines of this project
- [ ] I have followed all the instructions described in the initial task (check Definitions of Done)
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added reference to a related issue in the repository
- [ ] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: packages/good-design/src/apps/bridge/MicroBridgeStatus.tsx [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        </Popover>
+      )}
+    </HStack>
+    <Flex flex="1 1" alignItems="center" maxWidth="100%">
+      {txStatus && <ExplorerLink chainId={txStatus.chainId} addressOrTx={txStatus.transaction?.hash} />}
+    </Flex>
+  </Stack>
+);
+
+const PendingHeader = ({
```
Comment (L03TJ3) [inline packages/good-design/src/apps/bridge/MicroBridgeStatus.tsx:82]:
```suggestion
const StatusHeader = ({
```

File: packages/good-design/src/apps/bridge/MicroBridgeStatus.tsx [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+  const handleFinish = () => {
+    goToStep(0);
+  };
+
+  return (
+    <VStack p={4} space={4} width="100%" bgColor="goodWhite.100">
+      {relayStatus?.status === "Success" ? (
+        <Title variant="title-gdblue">{`Congratulations! Your bridging \n transaction has been successfully \n completed.`}</Title>
+      ) : null}
+      <PendingHeader
```
Comment (L03TJ3) [inline packages/good-design/src/apps/bridge/MicroBridgeStatus.tsx:176]:
```suggestion
      <StatusHeader
```

File: packages/good-design/src/core/layout/BalanceGD.tsx [hunk 3]
```diff
@@ -59,4 +59,73 @@ const BalanceGD: FC<BalanceGDProps> = ({ gdPrice, requiredChainId, refresh = "ne
...
+    ...(options || {})
+  };
+
+  const decimals = amount.currency.decimals;
+  const amountAbbr = abbreviateGd((Number(amount.value) / 10 ** decimals).toString());
+
+  return (
+    <HStack alignItems="flex-end">
+      <Text variant="l-grey-650" fontFamily="heading" color={color} {...props}>
+        {/* todo: awaiting decision on formatting */}
```
Comment (L03TJ3) [inline packages/good-design/src/core/layout/BalanceGD.tsx:119]:
```suggestion

```

File: packages/good-design/src/apps/bridge/MicroBridge.tsx [hunk 4]
```diff
@@ -205,67 +130,85 @@ export const MicroBridge = ({
...
-            {bridgeStatus?.status === "Success" && (
-              <StatusBox
-                text="Waiting for bridge relayers to relay to target chain... (Can take a few minutes)"
-                infoText="If you don't have enough native tokens on the target chain, a bridge relay service will execute the transfer for a small G$ fee"
-                txStatus={relayStatus}
-                sourceChain={sourceChain}
-              />
-            )}
+  return (
+    <VStack padding={4} width="100%" w="410px" alignSelf="center" backgroundColor="goodWhite.100">
```
Comment (sirpy) [inline packages/good-design/src/apps/bridge/MicroBridge.tsx:141]: no fixed width

File: packages/good-design/src/core/layout/BalanceGD.tsx [hunk 5]
```diff
@@ -59,4 +59,72 @@ const BalanceGD: FC<BalanceGDProps> = ({ gdPrice, requiredChainId, refresh = "ne
   );
 };
 
+const abbreviateGd = (numberStr: string, decPlaces = 2) => {
```
Comment (sirpy) [inline packages/good-design/src/core/layout/BalanceGD.tsx:62]: i believe there's already a utility we use for that

File: packages/good-design/src/utils/string.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+export const capitalizeFirstLetter = (str: string) => {
```
Comment (sirpy) [inline packages/good-design/src/utils/string.ts:1]: why not use styles for that?

**Review by sirpy — CHANGES_REQUESTED** [review]

File: packages/good-design/src/core/layout/BalanceGD.tsx (hunk 5 above)
Comment (L03TJ3) [reply packages/good-design/src/core/layout/BalanceGD.tsx:62]: The only thing I found was in the wallet which used an external library 'numeral' which I found a bit of an overkill

File: packages/good-design/src/utils/string.ts (hunk 6 above)
Comment (L03TJ3) [reply packages/good-design/src/utils/string.ts:1]: I am not sure I understand. what do you mean 'use styles' ?

File: packages/good-design/src/apps/bridge/MicroBridge.tsx (hunk 4 above)
Comment (L03TJ3) [reply packages/good-design/src/apps/bridge/MicroBridge.tsx:141]: ye, should not be there. removed

File: packages/good-design/src/utils/string.ts (hunk 6 above)
Comment (L03TJ3) [reply packages/good-design/src/utils/string.ts:1]: good question. no-idea why I did not think of it. changed now

File: packages/good-design/src/apps/bridge/MicroBridgeController.tsx [hunk 7]
```diff
@@ -33,16 +44,15 @@ const useCanBridge = (chain: "fuse" | "celo", amountWei: string) => {
...
-  const historySorted = sortBy((fuseHistory || []).concat(celoHistory || []), _ => _.data.from === "account");
+const BridgeHistoryWithRelay = () => {
+  const { historySorted } = useBridgeHistory() ?? {};
 
   const relayTx = useRelayTx();
   const [relaying, setRelaying] = useState<{ [key: string]: boolean }>({});
 
   const triggerRelay = useCallback(
     async (i: any) => {
+      console.log("relay triggered", { i });
```
Comment (sirpy) [inline packages/good-design/src/apps/bridge/MicroBridgeController.tsx:55]: remove all unrequired console logs

File: packages/good-design/src/apps/bridge/MicroBridge.tsx [hunk 8]
```diff
@@ -134,48 +72,40 @@ export const MicroBridge = ({
...
+    setPendingTransaction({ bridgeWeiAmount, expectedToReceive });
     onBridgeStart?.();
-
-    try {
-      await onBridge(inputWei, sourceChain);
-    } finally {
-      setBridging(false);
-    }
-  }, [setBridging, onBridgeStart, onBridge, inputWei, sourceChain]);
+    void nextStep();
```
Comment (sirpy) [inline packages/good-design/src/apps/bridge/MicroBridge.tsx:107]: shouldnt this be a condition that the onBridgeStart actually did some kind of TX?
what happens if the user rejects the bridge TX on metamask?

File: packages/good-design/src/apps/bridge/MicroBridge.tsx [hunk 9]
```diff
@@ -205,67 +135,113 @@ export const MicroBridge = ({
...
+          color: "white",
+          fontFamily: "subheading"
+        }}
+      />
+      <VStack space={1} mt={4} textAlign="left" width="100%">
+        <Text fontFamily="subheading" fontSize="xs" color="goodGrey.400">
+          Minimum amount to bridge: 1,000 G$
+        </Text>
+        <Text fontFamily="subheading" fontSize="xs" color="goodGrey.400">
+          Bridge Fee: <b>G$ 10 or 0.15%</b> of transaction <i>(See FAQs for more on Fees)</i>
```
Comment (sirpy) [inline packages/good-design/src/apps/bridge/MicroBridge.tsx:242]: shouldnt this be dynamic from the contract?

File: packages/good-design/src/apps/bridge/MicroBridgeController.tsx [hunk 10]
```diff
@@ -88,93 +99,238 @@ const MicroBridgeHistory = () => {
...
+
+  const { fee } = relayEvent?.data || {};
+
+  const date = moment(ethers.BigNumber.from(timestamp).toNumber() * 1000).utc();
+  const dateFormatted = date.format("DD.MM.YY");
+  const dateHours = date.format("HH:mm");
+
+  const targetChain = parseInt(targetChainId);
+  const sourceChain = targetChain === 122 ? 42220 : 122;
+  const test = 12223;
```
Comment (sirpy) [inline packages/good-design/src/apps/bridge/MicroBridgeController.tsx]: whats this?

File: packages/good-design/src/apps/bridge/MicroBridgeController.tsx [hunk 11]
```diff
@@ -88,93 +99,238 @@ const MicroBridgeHistory = () => {
...
+  const originChain = useState<"fuse" | "celo">(chainId === 122 ? "fuse" : "celo");
+
+  const onBridgeStart = useCallback(async () => {
+    const [inputWei] = inputTransaction;
+
+    try {
+      await sendBridgeRequest(inputWei, originChain[0]);
+    } catch (e) {
+      console.error("sendBridgeRequest failed", { e });
+      // onBridgeFailed();
```
Comment (sirpy) [inline packages/good-design/src/apps/bridge/MicroBridgeController.tsx]: what happens if user rejects tx?

File: packages/good-design/src/apps/bridge/useBalanceHook.ts [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+import { useEthers } from "@usedapp/core";
+import { IGoodDollar } from "@gooddollar/goodprotocol/types";
+import { useGetContract, useGetEnvChainId } from "@gooddollar/web3sdk-v2";
+import { Lock } from "async-await-mutex-lock";
+
+import { G$Amount } from "@gooddollar/web3sdk-v2";
+
+const pollLock = new Lock();
+
+const g$BalanceFromExplorer = async (account: string | undefined, contracts: string[]) => {
```
Comment (sirpy) [inline packages/good-design/src/apps/bridge/useBalanceHook.ts:12]: why this hook? didnt we have a way to get balances from blockchain?

File: packages/good-design/src/core/buttons/LearnButton.tsx [hunk 13]
```diff
@@ -29,6 +29,11 @@ const linksNew: Record<learnSources, links> = {
     link: "https://gooddollar.notion.site/What-is-signing-b0019fe6c43241068050c9aa16e87ee1",
     label: "What is signing?",
     icon: BillySign
+  },
+  identity: {
+    link: "https://www.notion.so/gooddollar/Get-G-873391f31aee4a18ab5ad7fb7467acb3",
+    label: "Learn about the identification process",
+    icon: BillySign
```
Comment (sirpy) [inline packages/good-design/src/core/buttons/LearnButton.tsx:36]: is this related?

**Review by sirpy — CHANGES_REQUESTED** [review]

File: packages/good-design/src/apps/bridge/MicroBridgeController.tsx (hunk 10 above)
Comment (L03TJ3) [reply packages/good-design/src/apps/bridge/MicroBridgeController.tsx]: Already removed

File: packages/good-design/src/apps/bridge/MicroBridgeController.tsx (hunk 11 above)
Comment (L03TJ3) [reply packages/good-design/src/apps/bridge/MicroBridgeController.tsx]: Just added an error flow

File: packages/good-design/src/core/buttons/LearnButton.tsx (hunk 13 above)
Comment (L03TJ3) [reply packages/good-design/src/core/buttons/LearnButton.tsx:36]: Not for the functioning of the microbridge, but to release to protocolUI.

(New modals where added a while ago, the identity one never got updated)

File: packages/good-design/src/apps/bridge/useBalanceHook.ts (hunk 12 above)
Comment (L03TJ3) [reply packages/good-design/src/apps/bridge/useBalanceHook.ts:12]: Yes, but I did not want to use useDapp hooks (there almost no way to control their polling)

File: packages/good-design/src/apps/bridge/MicroBridge.tsx (hunk 8 above)
Comment (L03TJ3) [reply packages/good-design/src/apps/bridge/MicroBridge.tsx:107]: And the condition? No, we track the state of the transaction on the next step (with the sign / confirmation modals)

File: packages/good-design/src/apps/bridge/useBalanceHook.ts (hunk 12 above)
Comment (sirpy) [reply packages/good-design/src/apps/bridge/useBalanceHook.ts:12]: what do you mean, we control their polling everywhere. in many of the hooks we have the polling interval

File: packages/good-design/src/apps/bridge/MicroBridge.tsx (hunk 8 above)
Comment (sirpy) [reply packages/good-design/src/apps/bridge/MicroBridge.tsx:107]: so what happens if the user actually reject the TX?
it will go to the nextStep()
what will happen then and how? the onError callback?

File: packages/good-design/src/apps/bridge/useBalanceHook.ts (hunk 12 above)
Comment (L03TJ3) [reply packages/good-design/src/apps/bridge/useBalanceHook.ts:12]: yes which triggers around 8 re-renders

File: packages/good-design/src/apps/bridge/MicroBridge.tsx (hunk 8 above)
Comment (L03TJ3) [reply packages/good-design/src/apps/bridge/MicroBridge.tsx:107]: https://github.com/GoodDollar/GoodWeb3-Mono/blob/42ab4b093318660dc45469dba788a902b00b0acd/packages/good-design/src/apps/bridge/MicroBridgeController.tsx#L316

File: packages/good-design/src/apps/bridge/useBalanceHook.ts (hunk 12 above)
Comment (L03TJ3) [reply packages/good-design/src/apps/bridge/useBalanceHook.ts:12]: Besides that, we only have one useG$Balance hook, which is for G$ / GOOD / GDX
which also fetches for mainnet.

It was either refactoring hook / adding new hook / using explorer.

I like the flow with the explorer more (There is no need to know the balance every x blocks)

File: packages/good-design/src/apps/bridge/MicroBridgeController.tsx (hunk 7 above)
Comment (L03TJ3) [reply packages/good-design/src/apps/bridge/MicroBridgeController.tsx]: done here: https://github.com/GoodDollar/GoodWeb3-Mono/pull/166/commits/a7bf359545be2e4dedc0e2f0b36fd9edf7e83075
